compiler/Makefile

```diff
@@ -123,6 +123,19 @@ TARGETS = $(CHPL)
 
 LIBS = -lm
 
+ifeq ($(CHPL_MAKE_HOST_MEM), jemalloc)
+ifeq ($(CHPL_MAKE_PLATFORM), darwin)
+JEMALLOC_LIBS=-L$(shell jemalloc-config --libdir) -ljemalloc $(shell jemalloc-config --libs)
+else
+ifeq ($(CHPL_LIB_PIC), jemalloc)
```

Comment:
Does lib_pic impact our host builds or just target? (I'm not sure myself)

Reply:
Yeah I'm only seeing for target builds, will fix

Reply:
I think we only use it for target builds, and I think that's fine. I don't know that we need PIC support for the compile (and if we only support for target, then I think we can remove the PIC linking here)